aws_lambda_powertools/metrics/provider/base.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Tuple
+
+from aws_lambda_powertools.metrics.types import MetricSummary
+
+
+class MetricsProvider(ABC):
+    """Class for metric provider template
+
+    Use this template to create your own metric provider.
+
+    """
+
+    # MAX_METRICS controls how many metrics can be collected before auto-flushing
+    MAX_METRICS = 100
+    # MAX_DIMENSIONS controls how many dimension could be added
+    MAX_DIMENSIONS = 29
+    # use single metric when recording coldstart
+    Enable_Single_Metrics = True
+    # enable validation before sending out metrics. When enabled, validation will raise error if failed.
+    Validate_Metrics = False
```

Comment:
Roger, according to this implementation we have to replicate these constants within each provider class we create, correct? I think this implementation makes sense, but I was wondering if there is any case where these values could be dynamic and passed by the developer in the instance of the provider class. 

I say this because when we talk about CloudWatch EMF we have control over it, we know that we can flush every 100 metrics, for example and there is no overhead, but I'm wondering if there is any provider where the developer will want to flush metrics dynamically due to tcp packet size, some batch process that needs to have real-time metrics or other situations.

I'm just guessing about it and would love to hear from you what you think or if you've found something like this in your searches for other providers.